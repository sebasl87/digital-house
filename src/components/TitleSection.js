import React from "react";
import { View, Text } from "react-native";

import { styles } from "../styles";

const TitleSection = ({ text }) => {
  return (
    <View
      style={{ marginVertical: 20, alignItems: "flex-start", width: "100%" }}
    >
      <Text style={styles.titleSection}>{text}</Text>
    </View>
  );
};

export default TitleSection;
