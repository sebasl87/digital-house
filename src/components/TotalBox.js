import React from "react";
import { View, Text } from "react-native";

import { styles } from "../../styles";

const TotalBox = ({ points }) => {
  return (
    <View
      style={{
        alignItems: "center",
        width: "100%",
      }}
    >
      <View style={styles.blueBox}>
        <Text style={styles.monthBlueBox}>Diciembre</Text>
        <View style={styles.totalContainerBox}>
          <Text style={styles.totalBlueBox}>
            {points && points.toString()}.00 pts
          </Text>
        </View>
      </View>
    </View>
  );
};

export default TotalBox;
