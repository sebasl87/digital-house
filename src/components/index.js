export { default as TitleSection } from "./TitleSection";
export { default as TotalBox } from "./TotalBox";
