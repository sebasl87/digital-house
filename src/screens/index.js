export { default as HomeScreen } from "./HomeScreen";
export { default as ProductDetail } from "./ProductDetail";
