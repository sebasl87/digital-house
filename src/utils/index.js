import { MONTHS } from "../constants";

export const getTotal = (arr) => {
  const sum = arr
    .filter((mov) => mov.is_redemption)
    .reduce(function (prev, current) {
      return prev + +current.points;
    }, 0);
  const rest = arr
    .filter((mov) => mov.is_redemption === false)
    .reduce(function (prev, current) {
      return prev + +current.points;
    }, 0);
  const result = (sum - rest).toLocaleString("en");
  return result;
};

export const formatDate = (fecha) => {
  const dateFormat =
    fecha.slice(8, 10) +
    " de " +
    MONTHS[fecha.slice(5, 7)] +
    ", " +
    fecha.slice(0, 4);

  return dateFormat;
};
