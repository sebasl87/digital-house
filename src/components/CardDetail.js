import React from "react";
import { View, Text } from "react-native";

import { styles } from "../../styles";
import { formatDate } from "../utils";
import TitleSection from "./TitleSection";

const MovTable = ({ product, points, isRedemption, date }) => {
  return (
    <View style={styles.containerProductInfo}>
      <TitleSection text="Detalles del producto" />
      <Text style={[styles.monthBlueBox, { color: "#000" }]}>
        Comprado el {formatDate(date)}
      </Text>
      <TitleSection
        style={{ marginBottom: 32 }}
        text={
          isRedemption
            ? "Con esta compra utilizaste"
            : "Con esta compra acumulaste"
        }
      />
      <Text
        style={[
          styles.monthBlueBox,
          { color: "#000", fontSize: 24, lineHeight: 33 },
        ]}
      >
        {points} puntos
      </Text>
    </View>
  );
};

export default MovTable;
