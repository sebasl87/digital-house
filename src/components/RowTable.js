import { useNavigation } from "@react-navigation/native";
import React from "react";
import { View, Text, Image, TouchableOpacity } from "react-native";

import { styles } from "../../styles";
import { formatDate } from "../utils";

const RowTable = ({ product, points, isRedemption, date, image }) => {
  const navigation = useNavigation();

  return (
    <TouchableOpacity
      onPress={() => {
        navigation.navigate("Product", {
          product,
          points,
          isRedemption,
          date,
          image,
        });
      }}
    >
      <View style={styles.rowTable}>
        <Image
          source={{ uri: image }}
          style={{ width: 55, height: 55, borderRadius: 10 }}
        />
        <View style={{ marginLeft: 12, width: 200 }}>
          <Text
            style={[styles.titleSection, { color: "#000", marginVertical: 5 }]}
          >
            {product}
          </Text>
          <Text style={[styles.dateText, { marginTop: 2 }]}>
            {formatDate(date)}
          </Text>
        </View>
        <View
          style={{
            alignItems: "center",
            flexDirection: "row",
          }}
        >
          {isRedemption ? (
            <Text style={[styles.monthBlueBox, { color: "#FF0000" }]}>-</Text>
          ) : (
            <Text style={[styles.monthBlueBox, { color: "#00B833" }]}>+</Text>
          )}
          <Text style={[styles.monthBlueBox, { color: "#000" }]}>{points}</Text>
          <View style={{ justifyContent: "center", alignItems: "flex-end" }}>
            <Text
              style={[styles.monthBlueBox, { color: "#000", marginLeft: 8 }]}
            >
              &gt;
            </Text>
          </View>
        </View>
      </View>
    </TouchableOpacity>
  );};

export default RowTable;
