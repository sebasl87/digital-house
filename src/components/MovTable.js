import React from "react";
import { View } from "react-native";

import { styles } from "../../styles";
import RowTable from "./RowTable";

const MovTable = ({ movements }) => {
  return (
    <View style={styles.table}>
      {movements &&
        movements.map(({ product, createdAt, points, is_redemption }) => (
          <RowTable
            date={createdAt}
            isRedemption={is_redemption}
            product={product}
            points={points}
          />
        ))}
    </View>
  );
};

export default MovTable;
