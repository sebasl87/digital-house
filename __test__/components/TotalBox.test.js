import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { TotalBox } from "../../src/components";

describe("TotalBox component", () => {
  it("should render successfully with mock data and isRedemption = true", () => {
    const totalBox = renderer.create(<TotalBox points={100} />).toJSON();

    expect(totalBox).toMatchSnapshot();
  });

  it("should render loader", () => {
    const totalBox = renderer.create(<TotalBox />).toJSON();

    expect(totalBox).toMatchSnapshot();
  });
});
