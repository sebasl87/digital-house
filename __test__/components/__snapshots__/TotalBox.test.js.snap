// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalBox component should render successfully with mock data and isRedemption = true 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#334FFA",
        "borderRadius": 20,
        "height": 143,
        "paddingLeft": 18,
        "paddingRight": 18,
        "paddingTop": 21,
        "shadowOffset": {
          "height": 6,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 4,
        "width": 286,
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFF",
          "fontFamily": "Avenir",
          "fontSize": 16,
          "fontStyle": "normal",
          "fontWeight": "800",
          "lineHeight": 22,
        }
      }
    >
      Diciembre
    </Text>
    <View
      style={
        {
          "alignItems": "flex-start",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFF",
            "fontFamily": "Avenir",
            "fontSize": 32,
            "fontStyle": "normal",
            "fontWeight": "800",
            "justifyContent": "center",
            "lineHeight": 44,
            "marginTop": 7,
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        100
        .00 pts
      </Text>
    </View>
  </View>
</View>
`;
